src/engraving/libmscore/textedit.cpp

```diff
@@ -243,33 +243,115 @@ void TextBase::insertText(EditData& ed, const String& s)
 
 bool TextBase::isEditAllowed(EditData& ed) const
 {
+    // Keep this method closely in sync with TextBase::edit()!
+
     if (ed.key == Key_Shift || ed.key == Key_Escape) {
         return false;
     }
 
+    bool ctrlPressed  = ed.modifiers & ControlModifier;
     bool shiftPressed = ed.modifiers & ShiftModifier;
-    if (shiftPressed && ed.key == Key_F2) {
-        return false; // Toggle Special Characters dialog
-    }
-
-    bool ctrlPressed = ed.modifiers & ControlModifier;
-
-    if (ctrlPressed && !shiftPressed) {
-        static std::set<int> ignore = {
-            Key_B,
-            Key_C,
-            Key_I,
-            Key_U,
-            Key_V,
-            Key_X,
-            Key_Y,
-            Key_Z
-        };
+    bool altPressed = ed.modifiers & AltModifier;
 
-        return !mu::contains(ignore, ed.key);
+    // Hex
+    if (ed.modifiers == (ControlModifier | ShiftModifier | KeypadModifier)) {
+        switch (ed.key) {
+        case Key_0:
+        case Key_1:
+        case Key_2:
+        case Key_3:
+        case Key_4:
+        case Key_5:
+        case Key_6:
+        case Key_7:
+        case Key_8:
+        case Key_9:
+            return true;
+        default: break;
+        }
+    } else if (ed.modifiers == (ControlModifier | ShiftModifier)) {
+        switch (ed.key) {
+        case Key_A:
+        case Key_B:
+        case Key_C:
+        case Key_D:
+        case Key_E:
+        case Key_F:
+            return true;
+        default: break;
+        }
     }
 
-    return true;
+    switch (ed.key) {
+    // Basic editing / navigation
+    case Key_Enter:
+    case Key_Return:
+    case Key_Delete:
+    case Key_Backspace:
+    case Key_Left:
+    case Key_Right:
+    case Key_Up:
+    case Key_Down:
+    case Key_Home:
+    case Key_End:
+    case Key_Tab:
+    case Key_Space:
+    case Key_Minus:
+    case Key_Underscore:
+        return true;
+    // Select all is handled by us
+    case Key_A:
+        if (ctrlPressed && !shiftPressed) {
+            return true;
+        }
+        break;
+    // Several commands
+    // TODO: looks like we're hard-coding keyboard shortcuts here, but what if user has edited those?
```

Comment:
I guess we should disable the ability to edit them in the preferences. There is no real way to change these shortcuts because they are hardcoded in TextBase and its derived classes. It would take a lot of refactoring to add a real ability to change these shortcuts, which would lead to a lot of regressions. So I don't think it's worth our effort. So instead of adding this comment, I would suggest opening an issue about removing these shortcuts from the preferences